### PR TITLE
[Chore](topn) split variables of topn

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
 split variables of topn

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -2408,8 +2424,8 @@ public int getAnalyzeTimeoutS() {
         return analyzeTimeoutS;
     }
 
-    public void setEnableTwoPhaseReadOpt(boolean enable) {
-        enableTwoPhaseReadOpt = enable;
+    public void disableTwoPhaseReadOpt() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:2427]: we should use `enable` as function or variable name, disable is hard to understand

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/DeferMaterializeTopNResult.java [hunk 2]
```diff
@@ -53,29 +53,25 @@ public class DeferMaterializeTopNResult implements RewriteRuleFactory {
...
+                                .when(r -> r.child().getLimit() < getTwoPhaseReadLimitThreshold())
                                 .whenNot(r -> r.child().getOrderKeys().isEmpty())
                                 .when(r -> r.child().getOrderKeys().stream().map(OrderKey::getExpr)
                                         .allMatch(Expression::isColumnFromTable))
                                 .when(r -> r.child().child().child().getTable().getEnableLightSchemaChange())
-                                .when(r -> r.child().child().child().getTable().isDupKeysOrMergeOnWrite())
-                                .then(r -> {
+                                .when(r -> r.child().child().child().getTable().isDupKeysOrMergeOnWrite()).then(r -> {
                                     LogicalFilter<LogicalOlapScan> filter = r.child().child();
                                     return deferMaterialize(r, r.child(), Optional.of(filter), filter.child());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/DeferMaterializeTopNResult.java:73]:
```suggestion
                                .when(r -> r.child().child().child().getTable().isDupKeysOrMergeOnWrite())
                                .then(r -> {
                                    LogicalFilter<LogicalOlapScan> filter = r.child().child();
                                    return deferMaterialize(r, r.child(), Optional.of(filter), filter.child());
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java [hunk 3]
```diff
@@ -46,35 +46,33 @@ public class LimitAggToTopNAgg implements RewriteRuleFactory {
...
                 // limit -> agg to topn->agg
-                logicalLimit(logicalAggregate())
-                        .when(limit -> ConnectContext.get() != null
-                                && ConnectContext.get().getSessionVariable().pushTopnToAgg
-                                && ConnectContext.get().getSessionVariable().topnOptLimitThreshold
-                                >= limit.getLimit() + limit.getOffset())
+                logicalLimit(logicalAggregate()).when(
+                        limit -> ConnectContext.get() != null && ConnectContext.get().getSessionVariable().pushTopnToAgg
+                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold >= limit.getLimit()
+                                        + limit.getOffset())
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java:52]:
```suggestion
                logicalLimit(logicalAggregate())
                        .when(limit -> ConnectContext.get() != null
                                && ConnectContext.get().getSessionVariable().pushTopnToAgg
                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold
                                >= limit.getLimit() + limit.getOffset())
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java [hunk 4]
```diff
@@ -46,35 +46,33 @@ public class LimitAggToTopNAgg implements RewriteRuleFactory {
...
                 logicalLimit(logicalProject(logicalAggregate()))
                         .when(limit -> ConnectContext.get() != null
                                 && ConnectContext.get().getSessionVariable().pushTopnToAgg
-                                && ConnectContext.get().getSessionVariable().topnOptLimitThreshold
-                                >= limit.getLimit() + limit.getOffset())
-                        .when(limit -> outputAllGroupKeys(limit, limit.child().child()))
-                        .then(limit -> {
+                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold >= limit.getLimit()
+                                        + limit.getOffset())
+                        .when(limit -> outputAllGroupKeys(limit, limit.child().child())).then(limit -> {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java:64]:
```suggestion
                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold
                                >= limit.getLimit() + limit.getOffset())
                        .when(limit -> outputAllGroupKeys(limit, limit.child().child()))
                        .then(limit -> {
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java [hunk 5]
```diff
@@ -46,35 +46,33 @@ public class LimitAggToTopNAgg implements RewriteRuleFactory {
...
-                        .when(topn -> ConnectContext.get() != null
-                                && ConnectContext.get().getSessionVariable().pushTopnToAgg
-                                && ConnectContext.get().getSessionVariable().topnOptLimitThreshold
-                                >= topn.getLimit() + topn.getOffset())
+                // topn -> agg: add all group key to sort key, if sort key is prefix of group
+                // key
+                logicalTopN(logicalAggregate()).when(
+                        topn -> ConnectContext.get() != null && ConnectContext.get().getSessionVariable().pushTopnToAgg
+                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold >= topn.getLimit()
+                                        + topn.getOffset())
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/LimitAggToTopNAgg.java:75]:
```suggestion
                logicalTopN(logicalAggregate())
                        .when(topn -> ConnectContext.get() != null
                                && ConnectContext.get().getSessionVariable().pushTopnToAgg
                                && ConnectContext.get().getSessionVariable().pushLimitToAggThreshold
                                >= topn.getLimit() + topn.getOffset())
```

**Comment by BiteTheDDDDt** [comment]
run buildall

**Pull request closed without merging**